crc32.c

```diff
@@ -37,11 +37,16 @@ uint32_t crc32(uint32_t crc, uint8_t *buf, size_t len) {
         buf += 8;
     }
 
-    while (len) {
-        crc = __crc32b(crc, *buf);
-        len--;
-        buf++;
+    if (len & 4) {
```

Comment:
I now see that I used > in the loop, do you mind changing that to >= while you are at it?